tests/unit/packages/registration-schema/validations-test.ts

```diff
@@ -0,0 +1,363 @@
+                registrationResponseKey: 'q2SingleSelectOptional',
+                groupType: 'single-select-input',
+            },
+
+            // Multiple Select: Required
+            {
+                labelBlock: {
+                    blockType: 'question-label',
+                    displayText: 'Multi Select: Required',
+                    index: 9,
+                    schemaBlockGroupKey: 'q3',
+                },
+                inputBlock: {
+                    blockType: 'multi-select-input',
+                    index: 9,
+                    schemaBlockGroupKey: 'q3',
+                    registrationResponseKey: 'q3MultiSelectRequired',
+                    required: true,
+                },
+                optionBlocks: [
+                    {
+                        blockType: 'select-input-option',
+                        index: 10,
+                        schemaBlockGroupKey: 'q3',
+                        displayText: 'option1 required multi select',
+                    },
+                    {
+                        blockType: 'select-input-option',
+                        index: 11,
+                        schemaBlockGroupKey: 'q3',
+                        displayText: 'option2 required multi-select',
+                    },
+                ],
+                schemaBlockGroupKey: 'q3',
+                registrationResponseKey: 'q3MultiSelectRequired',
+                groupType: 'multi-select-input',
+            },
+            // Multi Select: Optional
+            {
+                labelBlock: {
+                    blockType: 'question-label',
+                    displayText: 'Multi Select: Optional',
+                    index: 12,
+                    schemaBlockGroupKey: 'q4',
+                },
+                inputBlock: {
+                    blockType: 'multi-select-input',
+                    index: 13,
+                    schemaBlockGroupKey: 'q4',
+                    registrationResponseKey: 'q4MultiSelectOptional',
+                    required: false,
+                },
+                optionBlocks: [
+                    {
+                        blockType: 'select-input-option',
+                        index: 14,
+                        schemaBlockGroupKey: 'q4',
+                        displayText: 'option1 optional multi-select',
+                    },
+                    {
+                        blockType: 'select-input-option',
+                        index: 15,
+                        schemaBlockGroupKey: 'q4',
+                        displayText: 'option2 optional multi-select',
+                    },
+                ],
+                schemaBlockGroupKey: 'q4',
+                registrationResponseKey: 'q4MultiSelectOptional',
+                groupType: 'single-select-input',
+            },
+        ];
+        const result = buildValidation(groups);
+        assert.equal(result.q1SingleSelectRequired.response.length, 2,
```

Comment:
Consider avoiding hardcoded strings here. Try things like `result[groups[0].registrationResponseKey]`.

Reply:
This would have been an annoyance for the person having to touch this file next, I'm sure. Thank you!